pkg/cli/alpha/config-gen/types.go

```diff
@@ -0,0 +1,192 @@
+	// Defaults to the directoring containing the KubebuilderConfigGen configuration file.
+	Directory string `json:"projectDirectory" yaml:"projectDirectory"`
+
+	// Name is the name of project and used to generate the component and role names
+	// Defaults to metadata.name
+	Name string `json:"projectName" yaml:"projectName"`
+
+	// Namespace is the namespace to run the project in.
+	// Defaults to spec.name-system
+	Namespace string `json:"namespace" yaml:"namespace"`
+
+	// Image is the container image to run in the controller-manager Deployment
+	Image string `json:"image" yaml:"image"`
+
+	// Components is a map of components to enable or disable.
+	// Defaults to enabling:  ["namespace", "controller-manager", "rbac", "crds"]
+	Components map[string]bool `json:"components" yaml:"components"`
+
+	// DisableAuthProxy if set to true will disable the auth proxy
+	DisableAuthProxy bool `json:"disableAuthProxy,omitempty" yaml:"disableAuthProxy,omitempty"`
+
+	// ConversionWebhooks is a map of kinds to enable conversion webhooks for
+	ConversionWebhooks map[string]bool `json:"conversionWebhooks,omitempty" yaml:"conversionWebhooks,omitempty"`
+
+	// Development contains development options
+	Development DevelopmentOptions `json:"developmentOptions,omitempty" yaml:"developmentOptions,omitempty"`
+
+	// ComponentConfigFilepath is the relative path to a component config file used to configure the
+	// controller-manager
+	ComponentConfigFilepath string `json:"componentConfigFilepath,omitempty" yaml:"componentConfigFilepath,omitempty"`
+}
+
+// KubebuilderConfigGenStatus is runtime status for the api configuration
+type KubebuilderConfigGenStatus struct {
+	CertCA string
+
+	CertKey string
+
+	ComponentConfigString string
+}
+
+// DevelopmentOptions defines options for development installation
+type DevelopmentOptions struct {
+	// GenerateCert will cause a self signed certificate to be generated and injected
+	// into the Webhook caBundles.
+	GenerateCert bool `json:"generateCert,omitempty" yaml:"generateCert,omitempty"`
+
+	// CertDuration sets the duration for the cert
+	CertDuration time.Duration `json:"certDuration,omitempty" yaml:"certDuration,omitempty"`
+}
+
+// Default defaults the values
+func (kp *KubebuilderConfigGen) Default() error {
+	// merge project.yaml into fc
+	if kp.Project != nil {
+		// For compabitility with the PROJECT file format
+		kp.Spec = *kp.Project
+	}
+
+	// Validate the input
+	if kp.Spec.Name == "" {
+		if kp.Name == "" {
+			return errors.Errorf("must specify Kubebuilder projectName field")
+		}
+		kp.Spec.Name = kp.Name
+	}
+	if kp.Spec.Namespace == "" {
+		if kp.Namespace != "" {
+			kp.Spec.Namespace = kp.Namespace
+		}
+		kp.Spec.Namespace = kp.Spec.Name + "-system"
+	}
+	if kp.Spec.Image == "" {
+		return errors.Errorf("must specify Kubebuilder image field")
+	}
+	if kp.Spec.Development.CertDuration == 0 {
+		d := time.Hour * 1
+		kp.Spec.Development.CertDuration = d
```

Comment:
Is there any reason why you are not doing this?
```suggestion
		kp.Spec.Development.CertDuration = time.Hour
```